src/item.cpp

```diff
@@ -309,6 +309,12 @@ item& item::ammo_unset()
     return *this;
 }
 
+item& item::set_damage( double qty )
```

Comment:
`set_damage` by design can never destroy an item - it only sets damage constrained by `min_damage` and `max_damage`. It is used by code that wants to set the `damage` of a new item (for example in item spawn). It also doesn't fire the callback.